### PR TITLE
PR - Refactoring redux

### PR DESCRIPTION
**Pull request by Ehugo2000**
[176339136](https://www.pivotaltracker.com/story/show/176339136)
### User Story
```
As a developer
In order to manage state
I would like to be able to use Redux state management tools
```
## Changes proposed in this pull request:
* Refactoring project and implementing Redux
## What I have learned working on this feature:
* Redux
## Packages/Gems added
* Redux 
* React-Redux
## Screenshots (Client)
Screenshot from 2021-01-05 16-07-05

### CODE REVIEW

File: src/components/DisplayArticle.jsx
Comment (tochman) [inline src/components/DisplayArticle.jsx]: I would always strive to use the same key/attribute name in your component.

File: src/components/DisplayArticle.jsx
Comment (tochman) [inline src/components/DisplayArticle.jsx]: You can keep it as it is, or shorten the two lines above to one using object destructuring to access the properties. Like this:

```js
const { specificArticle, errorMessage } = useSelector(state => state);
```
short and neat...

File: src/components/DisplayArticle.jsx
Comment (tochman) [inline src/components/DisplayArticle.jsx]: Not sure about the conditional here, but let it be for now...

File: src/components/DisplayArticlesList.jsx
Comment (tochman) [inline src/components/DisplayArticlesList.jsx]: A small detail, but you can omit the parenthesis if you only have one argument. That goes for all your functions. And again, I see no reason to change the name. I would go with:
```js
const { newsFeed }  = useSelector(state => state);
```

File: src/state/store/initialState.js
Comment (tochman) [inline src/state/store/initialState.js]: since you have an empty array here, then you can probably skip the conditional in your component that checking if there is something in it. It is entirely fine to try to iterate over an empty array. 

Also, I would add `spacificArticle` here as an empty object. That will also save you a few lines of code in your components.

**Review by tochman — COMMENTED** [review]
No big blockers. Great job

File: src/components/DisplayArticle.jsx
Comment (sealfury) [reply src/components/DisplayArticle.jsx]: does that mean that ```viewArticle```  should be renamed to ```displayArticle```?

Comment (FlisAnn) [reply src/components/DisplayArticle.jsx]: We had a discussion in the team and decided that this naming made it more clear for us

File: src/components/DisplayArticle.jsx
Comment (FlisAnn) [reply src/components/DisplayArticle.jsx]: Fixed

File: src/components/DisplayArticle.jsx
Comment (FlisAnn) [reply src/components/DisplayArticle.jsx]: Yes, we will probably fix this further down the line :)

File: src/components/DisplayArticlesList.jsx
Comment (FlisAnn) [reply src/components/DisplayArticlesList.jsx]: Fixed

File: src/state/store/initialState.js
Comment (FlisAnn) [reply src/state/store/initialState.js]: Added specificArticle to initialState

File: src/components/DisplayArticle.jsx
Comment (emtalen) [inline src/components/DisplayArticle.jsx]: unfortunately @salindholm it looks like the tabs are haunting you 😆

File: src/state/store/initialState.js
Comment (emtalen) [inline src/state/store/initialState.js]: remove all these extra lines

**Review by emtalen — COMMENTED** [review]
just a bit of formatting! I will be waiting until it is done so you guys can get it merged

**Review by emtalen — APPROVED** [review]

**Pull request merged**
